threads/src/main/java/org/jboss/as/threads/BoundedQueueThreadPoolResourceDefinition.java

```diff
@@ -75,7 +75,7 @@ public static BoundedQueueThreadPoolResourceDefinition create(String type, Threa
         return new BoundedQueueThreadPoolResourceDefinition(blocking, registerRuntimeOnly, type, poolNameBase, resolverPrefix, addHandler, removeHandler);
     }
 
-    private BoundedQueueThreadPoolResourceDefinition(boolean blocking, boolean registerRuntimeOnly,
+    protected BoundedQueueThreadPoolResourceDefinition(boolean blocking, boolean registerRuntimeOnly,
```

Comment:
Please add

```
/**
 * @deprecated This class is not designed for subclassing and having this constructor be accessible is a specific workaround for WFCORE-1623 that may be reverted at any time
 */
@Deprecated
```

This whole module needs a bit of a rethink, but there are too many higher priorities right now. Until that happens I think a workaround like this is ok for closely related code like the JCA subsystem, but I don't want other people using it.

A longer term possibility is to have the constructor (private again) take a Parameters object, and then have public static create methods to generate that Parameters object. Plus a create method that takes the Parameters object and calls the constructor.

But I don't know if that really gains much; a review of the whole library is needed. So no point pursuing that right now.